rust/main/utils/run-locally/src/invariants/termination_invariants.rs

```diff
@@ -283,6 +283,8 @@ pub fn provider_metrics_invariant_met(
     .iter()
     .sum::<u32>();
 
+    log!("Provider instance count: {}", provider_count);
+
     if provider_count == 0 {
```

Comment:
I think there'll be at least `expected_messages` providers since we create ephemeral ones to prep each metadata, so let's use that

Reply:
Yeah, but we made it a guage, so it will always be in flux. Once the provider gets dropped, it will decrease. I'm not sure we can make any guarantees that it will be some number

Reply:
hm good point. In that case we can have two gauges instead, one tracking how many providers were ever created and one for how many were ever dropped. Then we'd plot the difference while still being able to check for correctness in e2e. Wdyt?

Reply:
oh good idea :+1: